### PR TITLE
Rsch/indentation/multiple start regexs

### PR DESCRIPTION
**Pull request by ReneSchmelzer**
Indentation regexs were split to increase maintainability.

More indentation tests added, including for nested package/interface blocks as provided by one of the other pull requests #87.

- [X] create indentation tests (files) for every diagram type (better overview of what is covered and what is missing)
- [X] write indentation tests to check all diagrams
- [X] have better and correct indentation handling of note blocks

→ I think issue #28 can be closed?
 

What is the procedure to handle package version? Last time you did not increase internal version variable, or did you?

### CODE REVIEW

File: plantuml-mode.el [hunk 1]
```diff
@@ -342,9 +344,40 @@ Uses prefix (as PREFIX) to choose where to display it:
     (defvar plantuml-keywords-regexp (concat "^\\s *" (regexp-opt plantuml-keywords 'words)  "\\|\\(<\\|<|\\|\\*\\|o\\)\\(\\.+\\|-+\\)\\|\\(\\.+\\|-+\\)\\(>\\||>\\|\\*\\|o\\)\\|\\.\\{2,\\}\\|-\\{2,\\}"))
     (defvar plantuml-builtins-regexp (regexp-opt plantuml-builtins 'words))
     (defvar plantuml-preprocessors-regexp (concat "^\\s *" (regexp-opt plantuml-preprocessors 'words)))
-    (defvar plantuml-indent-regexp-start "^[ \t]*\\(\\(?:.*\\)?\s*\\(?:[<>.*a-z-|]+\\)?\s*\\(?:\\[[a-zA-Z]+\\]\\)?\s+if\s+.*\\|loop\s+.*\\|group\s+.*\\|par\s*$\\|opt\s+.*\\|alt\s+.*\\|else\\|note\s+over\\|note\sas\s.*\\|note\s+\\(\\(?:\\(?:button\\|left\\|right\\|top\\)\\)\\)\\(?:\s+of\\)?\\|\\(?:\\(abstract \\)?class\\|enum\\|package\\|database\\|frame\\|cloud\\|folder\\)\s+.*{\\|activate\s+.+\\)")
-    (defvar plantuml-indent-regexp-end "^[ \t]*\\(endif\\|else\\|end\\|end\s+note\\|.*}\\|deactivate\s+.+\\)")
 
+    (defvar plantuml-indent-regexp-block-start "^.*{\s*$"
+      "Indentation regex for all plantuml elements that might define a {} block.
+Plantuml elements like skinparam, rectangle, sprite, package, ….
```
Comment (skuro) [inline plantuml-mode.el]: Could you please use `etc.` instead of the horizontal ellipsis `U+2026`?

**Comment by ReneSchmelzer** [comment]
done

**Comment by skuro** [comment]
locally `make test` fails, but I've a hunch it might be due to my local setup. Letting it run on CI now

**Comment by skuro** [comment]
code merged in `develop`, tonight I'll merge to `master` and cut a release. Thanks so much!

**Comment by skuro** [comment]
Code is now on `master` (which triggers a `MELPA` release) as well as on the `v.1.2.9` tag (which triggers a `MELPA stable` release) 👍

**Comment by ReneSchmelzer** [comment]
Great thanks! There still is the defconst of `plantuml-mode-version' which is not set correctly. How shall the version of plantuml-mode be handled correctly?

**Comment by skuro** [comment]
whoopsie! not a big deal at the moment I think. My preferred way of managing the version would be to have a `make release` command that

- verifies we're on `master`
- verifies that there are newer commits on `develop`
- merges `develop` into `master`
- sets the new version number on the `defconst`
- creates a new tag
- pushes both `master` and the newly created tag

**Pull request merged**